File: backend/core-service/src/Fix.Application/UseCases/Orders/Commands/LinkOrderMandateCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Orders.Dtos;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Orders.Commands;

/// <summary>Vínculo a posteriori: liga uma boleta sem mandato a um mandato ativo, com justificativa (carimbo permanente).</summary>
[RequireRole(RoleCodes.UpdateOrder)]
public sealed record LinkOrderMandateCommand(Guid UserId, Guid OrganizationId, Guid Id, Guid MandateId, string Justification)
    : ICommand<OrderDto>, IOrganizationRequest;

File: frontend/bff/src/application/orders/commands/link-order-mandate.command.ts
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

/** Vínculo a posteriori: boleta sem mandato ligada a um mandato ativo, com justificativa (carimbo permanente). */
export interface LinkOrderMandateCommand {
  context: RequestContext;
  id: string;
  mandateId: string;
  justification: string;
}
